rust/emitter/src/emitter.rs

```diff
@@ -707,6 +712,24 @@ impl InstructionWriter {
         self.emit_op(Opcode::JumpTarget);
     }
 
+    pub fn bytecode_offset(&mut self) -> BytecodeOffset {
+        BytecodeOffset { offset: self.bytecode.len() }
+    }
+
+    pub fn patch_jump_target(&mut self, jumplist: Vec<BytecodeOffset>) {
+        let target_offset = self.bytecode.len();
+        for jump in jumplist {
+            let mut index = jump.offset;
+            let new_target = (target_offset - index) as i32;
+            // manually replace the temporary i32 jump target with the the new
+            // location.
+            for byte in &new_target.to_le_bytes() {
+                index = index + 1;
+                self.bytecode[index] = *byte;
+            }
```

Comment:
it would be nice to make this a separate function

Reply:
There is one: `LittleEndian::write_i32(&mut self.bytecode[index..index + 4], new_target);`.

 https://docs.rs/byteorder/1.3.2/byteorder/trait.ByteOrder.html#method.write_i32